ckanext/harvest/logic/schema.py

```diff
@@ -12,39 +12,82 @@
                                    )
 from ckan.logic.converters import convert_to_extras, convert_from_extras
 
-from ckan.lib.navl.validators import (ignore_missing,
-                                      not_empty,
-                                      ignore,
-                                      if_empty_same_as,
-                                      )
-
-from ckanext.harvest.logic.validators import (harvest_source_url_validator,
-                                              harvest_source_type_exists,
-                                              harvest_source_config_validator,
-                                              harvest_source_extra_validator,
-                                              harvest_source_frequency_exists,
-                                              dataset_type_exists,
-                                              harvest_source_convert_from_config,
-                                              harvest_source_id_exists,
-                                              harvest_job_exists,
-                                              harvest_object_extras_validator,
-                                              )
+from ckanext.harvest.logic.validators import (
+    harvest_source_url_validator,
+    harvest_source_type_exists,
+    harvest_source_config_validator,
+    harvest_source_extra_validator,
+    harvest_source_frequency_exists,
+    dataset_type_exists,
+    harvest_source_convert_from_config,
+    harvest_source_id_exists,
+    harvest_job_exists,
+    harvest_object_extras_validator,
+)
+ckan_version = int(h.ckan_version().split(".")[1])
+if ckan_version >= 9:
+    from ckan.lib.navl.validators import (
+        ignore_missing,
```

Comment:
These should not be imported directly from core (except `unicode_safe` in the comment above). but using `toolkit.get_valiadator()`, eg:

```python
ignore_missing =  toolkit.get_validator('ignore_missing')

```